au.org.access-nri/model/spack/environment/deployment/1-0-0.json

```diff
@@ -0,0 +1,307 @@
+                        {
+                          "type": "number"
+                        }
+                      ]
+                    }
+                  },
+                  "target": {},
+                  "compiler": {},
+                  "buildable": {
+                    "type": "boolean",
+                    "default": true
+                  },
+                  "permissions": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                      "read": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "write": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "group": {
+                        "type": "string"
+                      }
+                    }
+                  },
+                  "package_attributes": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "patternProperties": {
+                      "\\w+": {}
+                    }
+                  },
+                  "providers": {},
+                  "variants": {
+                    "oneOf": [
+                      {
+                        "type": "string"
+                      },
+                      {
+                        "type": "array",
+                        "items": {
+                          "type": "string"
+                        }
+                      }
+                    ]
+                  },
+                  "externals": {
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "properties": {
+                        "spec": {
+                          "type": "string"
+                        },
+                        "prefix": {
+                          "type": "string"
+                        },
+                        "modules": {
+                          "type": "array",
+                          "items": {
+                            "type": "string"
+                          }
+                        },
+                        "extra_attributes": {
+                          "type": "object"
+                        }
+                      },
+                      "additionalProperties": true,
+                      "required": [
+                        "spec"
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "specs": {
+            "type": "array",
```

Comment:
No, we don't, but we can't say "make this a string" because I don't think the actual `spack.yaml` supports a single string spec, only an array of specs. Also, the CI expects the spec to live in `spack.specs[0]`. I've made a condition that states we can only have a min/max of 1.